File: src/estruture/bridge.py
# -*- coding: utf-8 -*-
from abc import ABC, abstractmethod

# Implementação
class JanelaImplementada(ABC):
    @abstractmethod
    def desenhar_janela(self, titulo):
        pass


class JanelaWindows(JanelaImplementada):
    def desenhar_janela(self, titulo):
        print(f"Desenhando janela no Windows: {titulo}")


class JanelaLinux(JanelaImplementada):
    def desenhar_janela(self, titulo):
        print(f"Desenhando janela no Linux: {titulo}")


# Abstração
class Janela:
    def __init__(self, implementacao: JanelaImplementada):
        self.implementacao = implementacao

    def desenhar(self, titulo):
        self.implementacao.desenhar_janela(titulo)


# Cliente
win = Janela(JanelaWindows())
win.desenhar("Configurações")

linux = JanelaLinux()
linux.desenhar("Preferências")

File: src/cpp/estructure/bridge/Bridge.cpp
#include "Bridge.hpp"
#include <iostream>

void JanelaWindows::desenhar() {
    std::cout << "Desenhando Janela Windows" << std::endl;
}

void JanelaLinux::desenhar() {
    std::cout << "Desenhando Janela Linux" << std::endl;
}

Janela::Janela(std::unique_ptr<IJanela> janela): janela_(std::move(janela)) {

}

void Janela::desenhar() const {
    janela_->desenhar();
}